### PR TITLE
 Refactor unpublishing presenter

### PR DESCRIPTION
**Pull request by suzannehamilton**
Extract a separate `ElasticsearchDeletePresenter` from the `ElasticsearchPresenter` to handle document unpublishing. Only the DeletePresenter now contains logic for looking up the existing document.

This will make it easier to implement unpublishing for external content, which also needs to look up the existing document.

This also takes us one step closer to refactoring this code into format-specific presenters, which we're planning to do soon.

https://trello.com/c/mwqMqK8s/542-allow-external-content-to-be-unpublished

It'll be easier to review the commits separately because the first one renames a file and lots of variables.

### CODE REVIEW

File: lib/govuk_index/document_type_inferrer.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module GovukIndex
-  class DocumentTypeInferer
+  class DocumentTypeInferrer
```
Comment (brenetic) [inline lib/govuk_index/document_type_inferrer.rb:2]: :man_facepalming:

**Comment by brenetic** [comment]
This looks cool. Is there a way we can DRY the identifier? It's not likely to ever change is it?

File: lib/govuk_index/document_type_inferrer.rb (hunk 1 above)
Comment (MatMoore) [reply lib/govuk_index/document_type_inferrer.rb:2]: [At least it's easy to change](https://www.quora.com/What-is-the-origin-of-HTTP-header-%E2%80%9Creferer%E2%80%9D-misspelling) 😁

**Review by MatMoore — APPROVED** [review]
LGTM

**Comment by MatMoore** [comment]
I've just rebased this so it can be merged. @brenetic I'll have a look at the identifier part and maybe raise another PR for it.

**Pull request merged**